packages/core/src/modules/proofs/ProofsApi.ts

```diff
@@ -555,29 +560,58 @@ export class ProofsApi<PPs extends ProofProtocol[]> implements ProofsApi<PPs> {
    */
   public async sendProblemReport(options: SendProofProblemReportOptions): Promise<ProofExchangeRecord> {
     const proofRecord = await this.getById(options.proofRecordId)
-    if (!proofRecord.connectionId) {
-      throw new AriesFrameworkError(`No connectionId found for proof record '${proofRecord.id}'.`)
-    }
 
     const protocol = this.getProtocol(proofRecord.protocolVersion)
-    const connectionRecord = await this.connectionService.getById(this.agentContext, proofRecord.connectionId)
 
-    // Assert
-    connectionRecord.assertReady()
+    const requestMessage = await protocol.findRequestMessage(this.agentContext, proofRecord.id)
 
     const { message: problemReport } = await protocol.createProblemReport(this.agentContext, {
       proofRecord,
       description: options.description,
     })
 
-    const outboundMessageContext = new OutboundMessageContext(problemReport, {
-      agentContext: this.agentContext,
-      connection: connectionRecord,
-      associatedRecord: proofRecord,
-    })
+    if (proofRecord.connectionId) {
+      const connectionRecord = await this.connectionService.getById(this.agentContext, proofRecord.connectionId)
 
-    await this.messageSender.sendMessage(outboundMessageContext)
-    return proofRecord
+      // Assert
+      connectionRecord.assertReady()
+
+      const outboundMessageContext = new OutboundMessageContext(problemReport, {
+        agentContext: this.agentContext,
+        connection: connectionRecord,
+        associatedRecord: proofRecord,
+      })
+
+      await this.messageSender.sendMessage(outboundMessageContext)
+      return proofRecord
+    } else if (requestMessage?.service) {
```

Comment:
What if we want to send a problem report as the verifier? We'd send the message to ourselves in that case (as we're the ones that created the request). I think we shouldn't assume the role here, and instead infer it based on the state

Reply:
Added check that state is `ProofState.RequestReceived`